api.go

```diff
@@ -506,6 +686,10 @@ func (r *Raft) RemovePeer(peer ServerAddress) Future {
 // If nonzero, timeout is how long this server should wait before the
 // configuration change log entry is appended.
 func (r *Raft) AddVoter(id ServerID, address ServerAddress, prevIndex uint64, timeout time.Duration) IndexFuture {
+	if r.protocolVersion < 1 {
+		return errorFuture{ErrUnsupportedProtocol}
+	}
+
```

Comment:
This only works for protocol >0, which unfortunately needs to support both ways, so there's nothing we can enforce here. You need to be able to remove protocol version 1 servers by address and add the new protocol version 2 servers by ID, so there's a small burden on the app during that time to call this in the right way.